rules/sort-sets.ts

```diff
@@ -0,0 +1,48 @@
+import type { Options } from './sort-array-includes'
+
+import { createEslintRule } from '../utils/create-eslint-rule'
+import { jsonSchema, sortArray } from './sort-array-includes'
+
+type MESSAGE_ID = 'unexpectedSetsOrder'
+
+export default createEslintRule<Options, MESSAGE_ID>({
+  name: 'sort-sets',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Enforce sorted sets.',
+    },
+    fixable: 'code',
+    schema: [jsonSchema],
+    messages: {
+      unexpectedSetsOrder: 'Expected "{{right}}" to come before "{{left}}".',
+    },
+  },
+  defaultOptions: [
+    {
+      type: 'alphabetical',
+      order: 'asc',
+      ignoreCase: true,
+      groupKind: 'literals-first',
+    },
+  ],
+  create: context => ({
+    NewExpression: node => {
+      if (
+        node.callee.type === 'Identifier' &&
+        node.callee.name === 'Set' &&
+        node.arguments.length &&
+        (node.arguments[0]?.type === 'ArrayExpression' ||
+          (node.arguments[0]?.type === 'NewExpression' &&
+            'name' in node.arguments[0].callee &&
+            node.arguments[0].callee.name === 'Array'))
+      ) {
+        let elements =
+          node.arguments[0].type === 'ArrayExpression'
+            ? node.arguments[0].elements
+            : node.arguments[0].arguments
+        sortArray<MESSAGE_ID>(context, 'unexpectedSetsOrder', elements)
```

Comment:
Wouldn't it be better to copypaste the code?
In case there are differences in the rule in the future.

Reply:
I don't believe it is necessary:
The code in common manages the sorting of the array element which is common whether we use
- `new Set(['a', 'b', 'c'])`
- or `['a', 'b', 'c'].includes(...` / `new Array('a', 'b', 'c').includes(...`

The only thing that could change between the two rules are options which only impact when we choose to sort or not.